packages/edit-site/src/components/block-editor/use-site-editor-mode.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+import { FOCUSABLE_ENTITIES } from './constants';
+
+export default function useSiteEditorMode() {
```

Comment:
I still feel that this should just be inlined where it's used personally. there's no need for a custom hook.

Reply:
I'm not hugely attached to this hook and in the interests of getting this merged I'm happy to remove it.

However, the hook is not there as a utility. Rather it's been introduced as a means of codify and naming concepts in the domain. I appreciate there are only a few usages, and we could have inline variables that do a similar job, but I just feel it documents the code in a better way (and DRYs up the code as a bonus).

Reply:
I'm always skeptical about introducing more abstractions personally. It's not like I'm really attached to this one being removed but it just feels like a wrapper that "rename" selectors. I fail to see the added value for such a hook.

Reply:
✅ In order to simply I've removed the hook and inlined the relevant code.